stix_shifter_modules/tanium/stix_transmission/connector.py

```diff
@@ -0,0 +1,133 @@
+        else:
+            current_query = f"limit={per_query_limit}&offset={current_offset}&expand=intelDoc"
+            
+        response_data = await self.query_tanium_api(current_query)
+        response_data_as_json = json.loads(response_data.content.decode('utf-8'))
+        self._add_results_to_final_dataset(response_data_as_json["data"])
+        return response_data_as_json["data"]
+        
+    def _add_results_to_final_dataset(self, current_query_results):
+        for batch_of_results in current_query_results:
+            batch_of_results["details"] = json.loads(batch_of_results["details"])
+            temp_file = batch_of_results["details"]["match"]["properties"]["file"]
+            batch_of_results["details"]["match"]["properties"]["file"]["hash"] = {"md5": temp_file["md5"], "sha1": temp_file["sha1"], "sha256": temp_file["sha256"]}
+            self.final_results.append(batch_of_results)
+        self.current_offset = self.current_offset + len(current_query_results)
+            
+
+    async def query_tanium_api(self, current_query):
+        response_dict = dict()
+        try:
+            response_dict = await self.api_client.call_api(current_query, 'GET', headers=self.headers, data={})
+        except Exception as err:
+            ErrorResponder.fill_error(self.return_obj, message=str(err), connector=self.connector)
+            raise
+
+        try:
+            response_code = response_dict.code
+            if response_code == 200:
+                return response_dict
+            else:
+                content_as_dict = ast.literal_eval(response_dict.content.decode("utf-8"))
+                if("Message" in content_as_dict):
+                    ErrorResponder.fill_error(self.return_obj, message=content_as_dict["Message"], connector=self.connector)
+                elif("errors" in content_as_dict):
+                    ErrorResponder.fill_error(self.return_obj, message=content_as_dict["errors"][0]["description"], connector=self.connector)
+                raise
+        except Exception:
+            self._handle_exceptions(response_dict, content_as_dict)
+            raise
+        
+    def _handle_exceptions(self, response_dict, content_as_dict):
+        if("code" in self.return_obj and self.return_obj["code"] == "unknown"):
+            error_message = f"tanium connector error => Error code|message : {str(response_dict.code)} | {str(responses[response_dict.code])}"
+            if("Message" in content_as_dict):
+                error_message = f"{error_message}. Request message : {content_as_dict['Message']}."
+                self.return_obj["error"] = error_message
+            elif("errors" in content_as_dict):
+                error_message = f"{error_message}. Request message : {content_as_dict['errors'][0]['description']}."
+                self.return_obj["error"] = error_message
+            
+            if(response_dict.code == 401 or response_dict.code == 407):
```

Comment:
you can move this error code mappings to stix_shifter_modules/tanium/stix_transmission/error_mapper.py